src/sig_stateful/sig_stfl.h

```diff
@@ -0,0 +1,224 @@
+/* Algorithm identifier for XMSSMT-SHA2_60/3_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_3_h60 "XMSSMT-SHA2_60/3_256"
+/* Algorithm identifier for XMSSMT-SHA2_60/6_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_6_h60 "XMSSMT-SHA2_60/6_256"
+/* Algorithm identifier for XMSSMT-SHA2_60/12_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_12_h60 "XMSSMT-SHA2_60/12_256"
+
+/**
+ * LMS will be added later. So far its only XMSS/XMSS^MT
+ */
+
+
+#define OQS_SIG_STFL_algs_length 20
+
+/**
+ * Returns identifiers for available signature schemes in liboqs.  Used with OQS_SIG_STFL_new.
+ *
+ * Note that algorithm identifiers are present in this list even when the algorithm is disabled
+ * at compile time.
+ *
+ * @param[in] i Index of the algorithm identifier to return, 0 <= i < OQS_SIG_algs_length
+ * @return Algorithm identifier as a string, or NULL.
+ */
+OQS_API const char *OQS_SIG_STFL_alg_identifier(size_t i);
+
+/**
+ * Returns the number of signature mechanisms in liboqs.  They can be enumerated with
+ * OQS_SIG_STFL_alg_identifier.
+ *
+ * Note that some mechanisms may be disabled at compile time.
+ *
+ * @return The number of signature mechanisms.
+ */
+OQS_API int OQS_SIG_STFL_alg_count(void);
+
+/**
+ * Indicates whether the specified algorithm was enabled at compile-time or not.
+ *
+ * @param[in] method_name Name of the desired algorithm; one of the names in `OQS_SIG_STFL_algs`.
+ * @return 1 if enabled, 0 if disabled or not found
+ */
+OQS_API int OQS_SIG_STFL_alg_is_enabled(const char *method_name);
+
+
+/**
+ * @brief SECRET KEY object for stateful signature schemes *
+ */
+typedef struct OQS_SECRET_KEY OQS_SECRET_KEY;
+
+typedef struct OQS_SECRET_KEY {
+
+	/** The (maximum) length, in bytes, of secret keys for this signature scheme. */
+	unsigned long long length_secret_key;
+
+	/** The physical secret key stored in memory as an array of bytes*/
+	uint8_t *secret_key;
+
+	unsigned long long (*sigs_total)(const OQS_SECRET_KEY *secret_key);
+
+	unsigned long long (*sigs_left)(const OQS_SECRET_KEY *secret_key);
+
+	OQS_STATUS (*lock_key)(OQS_SECRET_KEY *sk);
+	OQS_STATUS (*oqs_save_updated_sk_key)(const OQS_SECRET_KEY *sk);
+	OQS_STATUS (*release_key)(OQS_SECRET_KEY *sk);
+
+} OQS_SECRET_KEY;
+
+OQS_API OQS_SECRET_KEY *OQS_SECRET_KEY_new(const char *method_name);
+
+OQS_API void OQS_SECRET_KEY_free(OQS_SECRET_KEY *sk);
+
+/**
+ * Stateful signature scheme object
+ */
+typedef struct OQS_SIG_STFL {
+
+	/** Printable string representing the name of the signature scheme. */
+	const char *method_name;
+
+	/**
+	 * Printable string representing the version of the cryptographic algorithm.
+	 *
+	 * Implementations with the same method_name and same alg_version will be interoperable.
+	 * See README.md for information about algorithm compatibility.
+	 */
+	const char *alg_version;
+
+	/** The NIST security level (1, 2, 3, 4, 5) claimed in this algorithm's original NIST submission. */
+	uint8_t claimed_nist_level;
```

Comment:
How are we setting these for XMSS?  Since it's not a NIST submission perhaps this field doesn't make sense.

Reply:
I have removed this field from the SIG_STFL object.